README.md

```diff
@@ -12,14 +12,22 @@ The application uses server side rendering to generate the HTML pages.
 
 The application uses [Tailwind CSS](https://tailwindcss.com/) for styling.
 
-The application structure is a foray into Vertical Slice Architecture, where each capability (feature) is a separate package with one Kotlin source file each capability (create a party of characters, generate shared connections, etc.).
+## Vertical Slice Architecture
 
-Within each source file will be the route for the capability along with the "service" and repository functions needed to facilitate the capability. Models used by the service functions are stored in a `models` package under the `capability` package. Entity helpers (with the MongoDB collection name and helper functions to convert Kotlin data classes to and from Mongo Documents) are store outside the `capabilities` package, in a top level `db` package.
+The application structure is a foray into Vertical Slice Architecture, where each capability (feature) is a separate package with one Kotlin source file for each capability (create a party of characters, generate shared connections, etc.).
 
-The `plugins` package configures various aspects of the application including the HTTP Server, MongoDB connection, and the HTTP Client that interacts with Ollama.
+Within each capability source file, there will be the route for the capability, along with the "service" and repository functions needed to facilitate the capability. Models used by the service functions are meant to represent DDD entities and are stored in a `models` package under the `capability` package. Entity helpers (with the MongoDB collection name and helper functions to convert Kotlin data classes to and from Mongo Documents) are stored outside the `capabilities` package, in a top-level `db` package.
+
+The `plugins` package configures various aspects of the application, including the HTTP Server, MongoDB connection, and the HTTP Client that interacts with Ollama.
 
 The `web` package contains the `index` route and HTML elements along with a Styles CSS file that offers sets of Tailwind CSS classes for components.
 
+## Code Structure
+
+Looking at the route and service functions for each capability, the route is structured using a `try/catch` with any errors being returned as an error message and either a `404` when data is not found in Mongo or an HTTP status `400` error for exceptions.
+
+Service functions are wrapped in a `runCatching` and log exceptions in an `onFailure` block. `Result` objects are returned from service functions holding either data or the exception that occurred.
+
 TODO currently the connection to Ollama is done within an `Application` function in the plugins package and this needs to be moved elsewhere.
```

Comment:
**question (documentation):** Consider addressing the TODO item

This TODO item remains from the previous version. It might be worth considering whether this task should be addressed in a future update or if it should be tracked in a separate issue management system.